api/tasks_test.go

```diff
@@ -232,6 +268,42 @@ func TestTask_Constrain(t *testing.T) {
 	}
 }
 
+func TestTask_AddAffinity(t *testing.T) {
+	t.Parallel()
+	task := NewTask("task1", "exec")
+
+	// Add an affinity to the task
+	out := task.AddAffinity(NewAffinity("kernel.version", "=", "4.6", 100))
+	if n := len(task.Affinities); n != 1 {
```

Comment:
require.Len is handy for this https://godoc.org/github.com/stretchr/testify/require#Len